packages/credentials/src/compact-jwt.ts

```diff
@@ -0,0 +1,229 @@
+import type { CryptoAlgorithm, JwkParamsEcPrivate, JwkParamsOkpPrivate, PublicKeyJwk, Web5Crypto } from '@web5/crypto';
+import type { PortableDid } from '@web5/dids';
+
+import { PrivateKeyJwk, EdDsaAlgorithm, EcdsaAlgorithm  } from '@web5/crypto';
+import { DidDhtMethod, DidIonMethod, DidKeyMethod, DidResolver, utils } from '@web5/dids';
+import { JwtHeader, JwtPayload } from 'jwt-decode';
+import { Convert } from '@web5/common';
+
+/**
+ * Parameters for creating a JWT.
+ */
+export type CreateJwtParams = {
+  signerDid: PortableDid
+  payload: JwtPayload & Record<string, any>
+}
+
+/**
+ * Parameters for verifying a JWT.
+ */
+export type VerifyJwtParams = {
+  compactJwt: string
+}
+
+/**
+ * Parameters for decoding a JWT.
+ */
+export type DecodeJwtParams = {
+  compactJwt: string
+}
+
+/**
+ * Represents a verified JWT with signer information, header, and payload.
+ */
+export type VerifiedJwt = {
+  signerDid: string
+  header: JwtHeader
+  payload: JwtPayload & Record<string, any>
+}
+
+/**
+ * Represents a signer with a specific cryptographic algorithm and options.
+ * @template T - The type of cryptographic options.
+ */
+type Signer<T extends Web5Crypto.Algorithm> = {
+  signer: CryptoAlgorithm,
+  options?: T | undefined
+  alg: string
+  crv: string
+}
+
+const secp256k1Signer: Signer<Web5Crypto.EcdsaOptions> = {
+  signer  : new EcdsaAlgorithm(),
+  options : { name: 'ES256K'},
+  alg     : 'ES256K',
+  crv     : 'secp256k1'
+};
+
+const ed25519Signer: Signer<Web5Crypto.EdDsaOptions> = {
+  signer  : new EdDsaAlgorithm(),
+  options : { name: 'EdDSA' },
+  alg     : 'EdDSA',
+  crv     : 'Ed25519'
+};
+
+/**
+ * Class for handling Compact JSON Web Tokens (JWTs).
+ * This class provides methods to create, verify, and decode JWTs using various cryptographic algorithms.
+ */
```

Comment:
Is there a spec doc for compact JWTs we could link to here?

Reply:
[done](https://github.com/TBD54566975/web5-js/blob/verify-jwt/packages/credentials/src/compact-jwt.ts#L69)